vcd/datasource_vcd_tm_tier0_gateway.go

```diff
@@ -0,0 +1,71 @@
+package vcd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/go-vcloud-director/v3/govcd"
+	"github.com/vmware/go-vcloud-director/v3/types/v56"
+)
+
+const labelTmTier0Gateway = "TM Tier 0 Gateway"
+
+func datasourceVcdTmTier0Gateway() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: resourceVcdTmTier0GatewayRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: fmt.Sprintf("Display Name of %s", labelTmTier0Gateway),
+			},
+			"region_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: fmt.Sprintf("Parent %s ID", labelTmRegion),
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: fmt.Sprintf("Description of %s", labelTmTier0Gateway),
+			},
+			"parent_tier_0_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: fmt.Sprintf("Parent Tier 0 Gateway of %s", labelTmTier0Gateway),
+			},
+			"already_imported": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: fmt.Sprintf("Defines if the T0 is already imported of %s", labelTmTier0Gateway),
+			},
+		},
+	}
+}
+
+func resourceVcdTmTier0GatewayRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
Just a nit, but given there's no resource
```suggestion
func datasourceVcdTmTier0GatewayRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Reply:
Thanks!